File: apimsdc/MsdcQPrsList.cpp
/**
  * \file MsdcQPrsList.cpp
  * API code for table TblMsdcQPrsList (implementation)
  * \author Alexander Wirthmueller
  * \date created: 18 Dec 2018
  * \date modified: 18 Dec 2018
  */

#ifdef _WIN32
	#include "stdafx.h"
#endif

#include "MsdcQPrsList.h"

/******************************************************************************
 class MsdcQPrsList
 ******************************************************************************/

MsdcQPrsList::MsdcQPrsList(
			const uint jnum
			, const string Title
			, const string Firstname
			, const string Lastname
			, const string stubGrp
			, const string stubOwn
			, const string srefIxVSex
			, const string titIxVSex
			, const string telVal
			, const string emlVal
		) {
	this->jnum = jnum;
	this->Title = Title;
	this->Firstname = Firstname;
	this->Lastname = Lastname;
	this->stubGrp = stubGrp;
	this->stubOwn = stubOwn;
	this->srefIxVSex = srefIxVSex;
	this->titIxVSex = titIxVSex;
	this->telVal = telVal;
	this->emlVal = emlVal;
};

bool MsdcQPrsList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "MsdcQPrsList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "Firstname", "fnm", Firstname);
		extractStringUclc(docctx, basexpath, "Lastname", "lnm", Lastname);
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "srefIxVSex", "sex", srefIxVSex);
		extractStringUclc(docctx, basexpath, "titIxVSex", "sex2", titIxVSex);
		extractStringUclc(docctx, basexpath, "telVal", "tel", telVal);
		extractStringUclc(docctx, basexpath, "emlVal", "eml", emlVal);
	};

	return basefound;
};

/******************************************************************************
 class ListMsdcQPrsList
 ******************************************************************************/

ListMsdcQPrsList::ListMsdcQPrsList() {
};

ListMsdcQPrsList::ListMsdcQPrsList(
			const ListMsdcQPrsList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQPrsList(*(src.nodes[i]));
};

ListMsdcQPrsList& ListMsdcQPrsList::operator=(
			const ListMsdcQPrsList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQPrsList(*(src.nodes[i]));

	return *this;
};

ListMsdcQPrsList::~ListMsdcQPrsList() {
	clear();
};

void ListMsdcQPrsList::clear() {
	for (unsigned int i=0;i<nodes.size();i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListMsdcQPrsList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	MsdcQPrsList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListMsdcQPrsList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "MsdcQPrsList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i=0;i<ics.size();i++) {
		rec = new MsdcQPrsList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/MsdcQPrsList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japimsdc/MsdcQPrsList.java
/**
  * \file MsdcQPrsList.h
  * Java API code for record of table TblMsdcQPrsList
  * \author Alexander Wirthmueller
  * \date created: 18 Dec 2018
  * \date modified: 18 Dec 2018
  */

package apimsdc;

import org.w3c.dom.*;
import sbecore.*;

public class MsdcQPrsList {
	
	public MsdcQPrsList(
				int jnum
				, String Title
				, String Firstname
				, String Lastname
				, String stubGrp
				, String stubOwn
				, String srefIxVSex
				, String titIxVSex
				, String telVal
				, String emlVal
			) {
		this.jnum = jnum;
		this.Title = Title;
		this.Firstname = Firstname;
		this.Lastname = Lastname;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.srefIxVSex = srefIxVSex;
		this.titIxVSex = titIxVSex;
		this.telVal = telVal;
		this.emlVal = emlVal;
	};

	public int jnum;
	public String Title;
	public String Firstname;
	public String Lastname;
	public String stubGrp;
	public String stubOwn;
	public String srefIxVSex;
	public String titIxVSex;
	public String telVal;
	public String emlVal;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "MsdcQPrsList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			Firstname = Xmlio.extractStringUclc(doc, basexpath, "Firstname", "fnm", null, 0);
			Lastname = Xmlio.extractStringUclc(doc, basexpath, "Lastname", "lnm", null, 0);
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			srefIxVSex = Xmlio.extractStringUclc(doc, basexpath, "srefIxVSex", "sex", null, 0);
			titIxVSex = Xmlio.extractStringUclc(doc, basexpath, "titIxVSex", "sex2", null, 0);
			telVal = Xmlio.extractStringUclc(doc, basexpath, "telVal", "tel", null, 0);
			emlVal = Xmlio.extractStringUclc(doc, basexpath, "emlVal", "eml", null, 0);
			
			return true;
		};

		return false;
	};
};

